### ISSUE TITLE
Trying to aquire a lock on a key/value pair with a non existent session id returns a 500

### ISSUE THREAD

**Issue by dlouwers** [hashicorp/consul#1139]
When I call /v1/kv/whateverkey?acquire=<real_but_unknown_session_id> I get a 500 with 'For input string: "Invalid session"'.

This seems more like a 404 me. I think it would be great if this behaviour would be documented on https://www.consul.io/docs/agent/http/kv.html.

Best,

Dirk

**Comment by ryanuber** [comment]
Yep, the HTTP status codes are not the best in some places currently. Thanks for taking the time to report this. We are planning to go through them when the `/v2` endpoints are released in the future.

**Comment by dlouwers** [comment]
Thanks for the acknowledgement. Will go with the 500 + plain text message check for my v1 implementation.

**Comment by mjmasn** [comment]
@ryanuber same issue for executing a prepared query, [documentation](https://www.consul.io/docs/agent/http/query.html) says it returns a 404 but it's actually returning a 500.
